### ISSUE TITLE
Add typescript definitions

### ISSUE THREAD

**Issue by rricard** [apollostack/graphql-tag#1]
For now, in some use cases, TS (in atom) warns:  `Argument of type 'TemplateStringsArray' is not assignable to parameter of type 'string'.` for using gql like this: 

```js
gql` ... `
```

**Comment by rricard** [comment]
Found a fix to that: your typescript is too old. if you are using atom, please change the path to the typescriptServices in your config 
![settings______users_rricard_developer_github_com_apollostack_apollo-client](https://cloud.githubusercontent.com/assets/246520/16456608/c6f8df54-3e18-11e6-89fc-f98a08e07d1e.png)

**Comment by philcockfield** [comment]
@rricard - is this working for you?  I updated the `typescriptServices.js` file as you showed, to the version of TypeScript I'm using, but I'm still getting the build error:

    .../graphql.spec.tsx(26,22): error TS2345: Argument of 
    type 'TemplateStringsArray' is not assignable to parameter of 
    type 'string'.

![image](https://cloud.githubusercontent.com/assets/185555/16677858/e128bedc-452d-11e6-8134-d9852a040d8f.png)

Where are you getting your `typescriptServices.js` file from?

**Comment by rricard** [comment]
I installed the latest tsc from npm using `npm i -g typescript`.

**Comment by philcockfield** [comment]
@stubailo FYI here's the wrapper hack I wrote to get this working in TS:

    import gql from 'graphql-tag';
    import { Document } from 'graphql';



    /**
     * Processes the given template string parts.
     */
    function formatTemplate(literals: Array<string>, ...placeholders: Array<any>): any {
      const parts = [];
      literals.forEach((item, i) => {
        parts.push(item);
        if (placeholders[i]) {
          parts.push(placeholders[i]);
        }
      });
      return [parts.join('')];
    }



    /**
     * Process the given template-string into a GraphQL query.
     *  NOTE:
     *    This is a layer of indirection to `graphql-tag` because
     *    TypeScript does not play nicely with the module at the moment.
     */
    export default (literals: Array<string>, ...placeholders: Array<string>): Document => {
      const query = formatTemplate(literals, ...placeholders);
      return (<any> gql)(query);
    };

**Comment by philcockfield** [comment]
The type definition file that produces is:

    import { Document } from 'graphql';
    export declare function formatTemplate(literals: Array<string>, ...placeholders: Array<any>): any;
    declare var _default: (literals: string[], ...placeholders: string[]) => Document;
    export default _default;

**Comment by stubailo** [comment]
Hmm, how do dependencies in TypeScript definition files work? Since this package doesn't actually have a dependency on GraphQL, is it going to be weird?

**Comment by philcockfield** [comment]
I don't thinks so.  This is what you already have in your `index.d.ts`
https://github.com/apollostack/graphql-tag/blob/master/index.d.ts

    import { Document } from 'graphql';
    export default function gql(str: string): Document;

Where/how did you generate that file?

My understanding (as it stands) is that you don't need this package to reference `graphql`, however when someone installs this package within a TS project, they will need to add typings for graphql:

    typings install graphql --save

---
@stubailo, why not write this module in TS directly, like [Apollo-Client](https://github.com/apollostack/apollo-client)

**Comment by stubailo** [comment]
> Where/how did you generate that file?

I wrote it by hand!

> why not write this module in TS directly

Honestly it's too much work to set up the build process for a module this simple - it's like 10 lines of code.

**Comment by philcockfield** [comment]
Fair enough.

**Comment by stubailo** [comment]
Anyone have the time to write a better definition?

**Comment by philcockfield** [comment]
@stubailo - I suspect this would do the trick:

**index.d.ts**
```js
import { Document } from 'graphql';
export default function gql(literals: string[], ...placeholders: string[]): Document;
```

**Comment by stubailo** [comment]
OK I'll try this one and republish soon!

**Comment by mquandalle** [comment]
@stubailo I pushed the fix proposed by @philcockfield, which works as expected and fixes the OP bug. Feel free to publish on NPM if that’s good for you.

**Comment by stubailo** [comment]
Publishing now!

**Comment by stubailo** [comment]
Done - published as 0.1.10!
